### ISSUE TITLE
Accessibility - Incorrect focus order on MatDrawer when positioned 'end'

### ISSUE THREAD

**Issue by satheeshd** [angular/components#11992]
#### Bug, feature request, or proposal:
Bug/Feature request: 
Accessibility Hurdles: incorrect tab order when MatDrawer is positioned at the 'end'.`<mat-drawer [position]="'end'"></mat-drawer>`

#### What is the expected behavior?
**Scenario 1:**
If the `<mat-drawer-container></mat-drawer-container>` contains only a 'start' positioned MatDrawer, then the tab order should be as below:
first: `<mat-drawer [position]="'start'"></mat-drawer>`
second: `<mat-drawer-content></mat-drawer-content>`

**Scenario 2:**
If the `<mat-drawer-container></mat-drawer-container>` contains both 'start' positioned and 'end' positioned MatDrawers, then the tab order should be as below:
first: `<mat-drawer [position]="'start'"></mat-drawer>`
second: `<mat-drawer-content></mat-drawer-content>`
third: `<mat-drawer [position]="'end'"></mat-drawer>`

**Scenario 3:**
If the `<mat-drawer-container></mat-drawer-container>` contains only an 'end' positioned MatDrawer, then the tab order should be as below:
first: `<mat-drawer-content></mat-drawer-content>`
second: `<mat-drawer [position]="'end'"></mat-drawer>`

#### What is the current behavior?
If the <mat-drawer-container></mat-drawer-container> contains an 'end' positioned MatDrawer, then the tab order is incorrect as below:
first:  `<mat-drawer [position]="'end'"></mat-drawer>`
second:`<mat-drawer-content></mat-drawer-content>`

#### What are the steps to reproduce?
[Scenario 3](https://stackblitz.com/edit/angular-vzoef2?file=app%2Fsidenav-drawer-overview-example.html)
[Scenario 2](https://stackblitz.com/edit/angular-egcpfh?file=app/sidenav-drawer-overview-example.html)

#### What is the use-case or motivation for changing an existing behavior?
for accessibility & usability reasons, when the page content is in LTR mode, the tab order has to be from left content to right. Currently the end positioning of <mat-drawer> is achieved using the 'positioning right' strategy using 'css'. This causes the DOM elements still remain before the <mat-drawer-content></mat-drawer-content>items. This causes tab order related issues. The main content can be reached only after the 'end' positioned <mat-drawer>contents are reached.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Material: 6.3.1

#### Is there anything else we should know?
WCAG Recommendations:
[WCAG (2.4.3)](https://www.w3.org/TR/WCAG20/)
[Quick refrences](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-navigation-mechanisms-focus-order)

**Comment by crisbeto** [comment]
Closing as a duplicate of https://github.com/angular/components/issues/15247.
